src/Hox/Rendering.fs

```diff
@@ -278,78 +285,36 @@ module Chunked =
-          // when they're added to the parent node, so we don't need to
-          // do anything here.
+        | Text text, _ -> yield  cachedHtmlEncode text
+        | Raw raw, _ -> yield raw
+        | Comment comment, _ -> yield $"<!--{cachedHtmlEncode comment}-->"
+        | Fragment nodes, _ ->
+            let mutable node = nodes.Last
+            while node <> null do
+              dfs.Push(node.Value, false)
+              node <- node.Previous
+        | AsyncNode node, _ ->
           let! node = node cancellationToken
-          stack.Push(node, false, depth)
-
-        | AsyncSeqNode nodes ->
-          // This number is tricky, I've seen stack frames overflow at 248
-          // but also at 355, we'll keep it to 235 for now, we'll need to investigate
-          // further or if you're reading this and have an idea, let me know please :).
-          if depth > 235 then
-            // Similarly to the Builder module, we can't cancell this operation
-            // as it doesn't support cancellation, so we'll just have to wait
-            // for it to finish.
-            let! nodes = nodes |> TaskSeq.toArrayAsync
-
-            if nodes.Length > 0 then
-              for child = nodes.Length - 1 downto 0 do
-                stack.Push(nodes[child], false, depth)
-
-          else
-            // This part makes me a bit sad, since we can't reverse the sequence
-            // we have to add each node to a queue and then dequeue them in order
-            // to render them in the correctly in the final html string.
-            // so... we're kind of doing the same thing above but still yielding
-            // recursively the chunks once available Not fan of this approach
-            // we might as well just keep above's approach and that's it.
-
-            // Ideally I'd just for node in nodes do renderThing but that would
-            // render the nodes in reverse order.
-            // I guess we'll be stuck until recursion us supported for Tasks/TaskSeq in the F# compiler 🫠
-            // that or we could use async computations to attempt a recursive solution
-            // but that would potentially make ValueTasks Hot which we'd like to avoid untill
-            // we know the ValueTask is actually an async operation (see the cases above).
-            let items = LinkedList()
-
-            for node in nodes do
-              items.AddFirst(node) |> ignore
-
-            for item in items do
-              stack.Push(item, false, depth)
-            // The next renderNode call will check it's own
-            // cancellation token and throw if needed.
-            yield! renderNode(stack, cancellationToken)
+          dfs.Push(node, false)
+        | AsyncSeqNode nodes, _ ->
+          match! TaskSeq.headTail nodes cancellationToken with
+          | ValueSome (node, nodes) ->
+            dfs.Push(AsyncSeqNode nodes, false)
+            dfs.Push(node, false)
```

Comment:
This allows to handle a node as soon as it's ready, then we'll continue again to the tail nodes once it's complete.